### PR TITLE
[Harness] Add extra tests for the SimDevice and Simulator classes.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Add test that were not added before because the diff of the previous PR was too large. Test make sure that commands are executed as expected and the xml parsed correctly.

### CODE REVIEW

File: tests/xharness/Xharness.Tests/Hardware/Tests/SimDeviceTest.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using Xharness.Execution;
+using Xharness.Hardware;
+using Xharness.Logging;
+
+namespace Xharness.Tests.Hardware.Tests {
+	[TestFixture]
+	public class SImDeviceTest {
```
Comment (spouliot) [inline tests/xharness/Xharness.Tests/Hardware/Tests/SimDeviceTest.cs]: ^ alignment

File: tests/xharness/Xharness.Tests/Hardware/Tests/SimDeviceTest.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		[TestCase ("com.apple.CoreSimulator.SimRuntime.iOS-12-1", "iOS 12.1")]
+		[TestCase ("com.apple.CoreSimulator.SimRuntime.iOS-10-1", "iOS 10.1")]
+		public void OSVersionTest (string runtime, string expected)
+		{
+			simulator.SimRuntime = runtime;
+			Assert.AreEqual (expected, simulator.OSVersion);
+		}
+
+		[Test]
+		public async Task ErateAsyncTest ()
```
Comment (spouliot) [inline tests/xharness/Xharness.Tests/Hardware/Tests/SimDeviceTest.cs]: typo `Erase`

File: tests/xharness/Xharness.Tests/Hardware/Tests/SimDeviceTest.cs (hunk 1 above)
Comment (spouliot) [reply tests/xharness/Xharness.Tests/Hardware/Tests/SimDeviceTest.cs]: not so sure, it looks like a tab that did not use to be 4 spaces
maybe it's an GitHub change (related to .editorconfig) ?

Comment (mandel-macaque) [reply tests/xharness/Xharness.Tests/Hardware/Tests/SimDeviceTest.cs]: it is a tab. I have it setup to 4 and looks as expected. Funny .editorconfig side effect.

Comment (rolfbjarne) [inline tests/xharness/Xharness.Tests/Hardware/Tests/SimDeviceTest.cs]:
```suggestion
	public class SimDeviceTest {
```

Comment (rolfbjarne) [reply tests/xharness/Xharness.Tests/Hardware/Tests/SimDeviceTest.cs]: Looks like a GitHub bug... 2 space indent for the first indentation. Maybe they're counting the `+` and the space after that as indentation?

File: tests/xharness/Xharness.Tests/Hardware/Tests/SimulatorsTest.cs [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		public void TearDown ()
+		{
+			executionLog = null;
+			processManager = null;
+			simulators = null;
+		}
+
+
+		[TestCase (false)] // no timeout
+		[TestCase (true)] // timeoout
```
Comment (rolfbjarne) [inline tests/xharness/Xharness.Tests/Hardware/Tests/SimulatorsTest.cs]:
```suggestion
		[TestCase (true)] // timeout
```

**Pull request merged**